src/Simplify.elm

```diff
@@ -8777,6 +8777,17 @@ isEqualToSomethingFunction rawNode =
             Nothing
 
 
+expressionContainsVariable : String -> Expression -> Bool
+expressionContainsVariable variableToFind expressionNode =
+    case expressionNode of
+        Expression.FunctionOrValue [] variable ->
+            variable == variableToFind
+
+        nonVariableExpressionNode ->
+            List.any (\(Node _ sub) -> expressionContainsVariable variableToFind sub)
+                (AstHelpers.subExpressions nonVariableExpressionNode)
```

Comment:
This sounds like a pretty expensive function, because this is (unlikely but) potentially a large expression, and we'll be going through every node of it. But I don't have a better solution except abandoning this check when it looks to be this complex.
(So let's leave it as is :+1: )